_posts/wasmer-go-embedding-1.0.md

```diff
@@ -0,0 +1,534 @@
+
+Let's talk a moment about compilers and engines. Compilers aim at
+compiling the WebAssembly modules into executable codes. Engines drive
+the compilation _and_ the execution of the WebAssembly modules. This
+design provides a unique flexibility which allows Wasmer to be used in
+various contexts.
+
+### Cranelift has new companions: Singlepass and LLVM!
+
+The Wasmer runtime provides 3 compilers to compile the WebAssembly
+modules into executable codes:
+
+* Singlepass: Super fast compilation times, slow execution times. Not
+  prone to JIT-bombs,
+  
+* Cranelift: Fast compilation times, fast execution times,
+
+* LLVM: Slow compilation times, very fast execution times (close to
+  native).
+
+Previously, `wasmer-go` was providing only one compiler:
+Cranelift. However, a non-negligible population of our users run small
+WebAssemby modules. In that specific case, execution time is not an
+issue since it will always be quick, however we must improve the
+compilation time. And that's why we are happy to announce that
+`wasmer-go` now embeds the Singlepass compiler too by default!
+
+Double announce: When execution performance really matters for you,
+you can now use the LLVM compiler too! Even if the default
+`libwasmer`s embedded inside `wasmer-go` do not provide a support for
+LLVM _yet_ (we are working on it), the entire API already support
+it. It is really easy to use your own custom `libwasmer` and build
+against it (see below about the `custom_wasmer_runtime` tag).
+
+Cranelift will continue to be the default compiler. To change that
+behaviour, one needs to create a new configuration for the engine and
+use `Config.UseSinglepassCompiler` or `Config.UseLLVMCompiler`:
+
+```go
+// Configure the engine to use the Singlepass compiler.
+config := wasmer.NewConfig().UseSinglepassCompiler()
+engine := wasmer.NewEngineWithConfig(config)
+
+// Compile the WebAssembly module.
+store := wasmer.NewStore(engine)
+module, _ := wasmer.NewModule(store, wasmBytes)
+
+// Instantiate it, and something with it.
+instance, _ := wasmer.NewInstance(module, wasmer.NewImportObject())
+```
+
+It's that simple!
+
+Note: The new `IsCompilerAvailable` function might be your best friend to
+test whether a compiler is available.
+
+### JIT and Native engines
+
+The Wasmer runtime also provides 3 engines to compile _and_ to execute
+the WebAssembly modules. In a nutshell:
+
+* The JIT engine stores the executable code in memory,
+
+* The Native engine stores the executable code in a native shared
+  library object (`.so`, `.dylib`, or `.dll` files depending on the
+  Operating System it runs),
+  
+* The Object File engine, which isn't relevant in the context of Go,
+  we skip it ([learn
+  more](https://github.com/wasmerio/wasmer/tree/d1bd9eac00a3e1d445499e47d8cb3a632985e0c6/lib/engine-object-file)).
+  
+Previously, `wasmer-go` was providing only the JIT engine. Now, it's
+our pleasure to announce that the Native engine is now part of the
+family!
```

Comment:
```suggestion
Previously, `wasmer-go` only provided the JIT engine. Now, it's
our pleasure to announce that the Native engine is now also part of the
family!
```

Reply:
"also" is optional, unsure if it's actually better or not 🤔

Reply:
:+1: